src/components/connect.js

```diff
@@ -192,7 +196,14 @@ function connect(mapPropsToRequestsToProps, defaults, options) {
         this._unmounted = true
       }
 
+      shouldComponentUpdate(nextProps, nextState) {
+        return !options.pure ||
+          this.state.data != nextState.data ||
+          (dependsOnProps ? (this._propsDidChange || this.props.children != nextProps.children) : !shallowEqual(this.props, nextProps))
```

Comment:
So writing a `componentShouldUpdate` for components that have children is kind of a bummer, because the children will always be different on every update, even if they didn't change. The only time this is not so is when the component doesn't have children, in which `this.props.children == nextProps.children` is always true. Since `connected` components might or might not have `children` this condition is there for the times when the wrapped component doesn't have children.

(You have to re-render a parent when the children change, otherwise the new children would never appear on the screen)